### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by asolntsev**
Probably it's easier to review this PR commit-by-commit.

This PR removes three static fields that were growing endlessly:
1. `BaseFaker.PROVIDERS_MAP`
2. `BaseFaker.CLASSES`
3. `FakeValuesService.REGEXP2SUPPLIER_MAP`

### CODE REVIEW

File: src/test/java/net/datafaker/service/FakeValuesServiceTest.java [hunk 1]
```diff
@@ -27,7 +27,8 @@
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeFormatterBuilder;
 import java.time.temporal.ChronoField;
-import java.util.*;
+import java.util.List;
+import java.util.Locale;
```
Comment (kingthorin) [inline src/test/java/net/datafaker/service/FakeValuesServiceTest.java:31]: thank you :wink:

Comment (asolntsev) [reply src/test/java/net/datafaker/service/FakeValuesServiceTest.java:31]: You are welcome :) 

P.S. Such import changes are usually done automatically by IDE.

Comment (kingthorin) [reply src/test/java/net/datafaker/service/FakeValuesServiceTest.java:31]: Ya I know, but it's a nit pick of mine so I'm happy to see it :wink:

**Review by kingthorin — APPROVED** [review]
Seems good to me. Thanks

**Comment by snuyanzin** [comment]
Thanks for working on this

I guess we also need to run some measurement tests to see what this change brings to us
I will run it today and post the result later

**Comment by asolntsev** [comment]
> we also need to run some measurement tests 

At least I run `Issue1178Test`, and it takes ~the same time in `main` branch and `fix/memory-leak`.

**Comment by snuyanzin** [comment]
there is no significant changes
probably on only one improvement for  `DatafakerSimpleMethods.numberBetween` about 10%, others are more or less same
before 
```
Benchmark                              Mode  Cnt       Score     Error   Units
DatafakerSimpleMethods.firstname      thrpt   10    9383.836 ± 110.342  ops/ms
DatafakerSimpleMethods.fullname       thrpt   10    3108.838 ±  27.933  ops/ms
DatafakerSimpleMethods.numberBetween  thrpt   10  116170.962 ± 705.899  ops/ms
DatafakerSimpleMethods.streetAddress  thrpt   10    2213.100 ±  13.656  ops/ms

Benchmark                        Mode  Cnt      Score      Error   Units
DatafakerExpressions.bothify    thrpt   10  13654.868 ±   78.549  ops/ms
DatafakerExpressions.letterify  thrpt   10  18601.883 ± 1715.311  ops/ms
DatafakerExpressions.numerify   thrpt   10  16980.816 ±  972.231  ops/ms
DatafakerExpressions.regexify   thrpt   10  19546.992 ± 2126.820  ops/ms


Benchmark                  Mode  Cnt    Score    Error   Units
DatafakerCsvjson.csvJson  thrpt   10  460.369 ± 44.570  ops/ms
```
after
```
===============================================================
Benchmark                              Mode  Cnt       Score     Error   Units
DatafakerSimpleMethods.firstname      thrpt   10    9612.673 ± 171.756  ops/ms
DatafakerSimpleMethods.fullname       thrpt   10    3137.267 ±  69.310  ops/ms
DatafakerSimpleMethods.numberBetween  thrpt   10  132792.389 ± 277.564  ops/ms
DatafakerSimpleMethods.streetAddress  thrpt   10    2258.682 ±  31.129  ops/ms

Benchmark                        Mode  Cnt      Score      Error   Units
DatafakerExpressions.bothify    thrpt   10  13748.220 ± 1365.136  ops/ms
DatafakerExpressions.letterify  thrpt   10  20836.751 ±  490.235  ops/ms
DatafakerExpressions.numerify   thrpt   10  19378.905 ± 2497.980  ops/ms
DatafakerExpressions.regexify   thrpt   10  21656.561 ±  271.221  ops/ms

Benchmark                  Mode  Cnt    Score    Error   Units
DatafakerCsvjson.csvJson  thrpt   10  479.511 ± 12.053  ops/ms
```

**Review by snuyanzin — APPROVED** [review]

**Comment by kingthorin** [comment]
Thank you!

**Pull request merged**